File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.py
class Solution:
    def countStudents(self, students: List[int], sandwiches: List[int]) -> int:
        eated_count=0
        while students:
            if students[0]==sandwiches[0]:
                students.pop(0)
                sandwiches.pop(0)
                eated_count=0
            else:
                students.append(students.pop(0))
                eated_count+=1
            if eated_count==len(students):
                break

        return len(students)
        

File: 1700-number-of-students-unable-to-eat-lunch/1700-number-of-students-unable-to-eat-lunch.java
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
    //     Queue<Integer> queue = new LinkedList<>();
    //     for (int student : students) {
    //         queue.offer(student);
    //     }
    //     int i = 0; 
    //     int eatedCount = 0;
    //     while (!queue.isEmpty()) {
    //         if (queue.peek() == sandwiches[i]) {
    //             queue.poll();   
    //             i++;            
    //             eatedCount = 0; 
    //         } else {
    //             queue.offer(queue.poll()); 
    //             eatedCount++;
    //         }
    //         if (eatedCount == queue.size()) {
    //             break;
    //         }
    //     }
    //     return queue.size();
    // }

    int count[] = new int[2];
    for(int student:students){
        count[student]++;
    }
    for(int sandwiche:sandwiches){
        if(count[sandwiche]==0){
            break;
        }
        count[sandwiche]--;
    }
    return count[0]+count[1];
}

}